### PR TITLE
Extract the Main Input PCollection in ParDos

### PR DESCRIPTION
**Pull request by tgroh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`.
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by tgroh** [comment]
R: @kennknowles

File: runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/ParDos.java [hunk 1]
```diff
@@ -146,6 +146,27 @@ public static ParDoPayload toProto(ParDo.MultiOutput<?, ?> parDo, SdkComponents
...
 
+  public static RunnerApi.PCollection getMainInput(
+      RunnerApi.PTransform ptransform, Components components) throws IOException {
+    checkArgument(
+        ptransform.getSpec().getUrn().equals(PAR_DO_PAYLOAD_URN),
+        "Unexpected payload type %s",
+        ptransform.getSpec().getUrn());
+    ParDoPayload payload = ptransform.getSpec().getParameter().unpack(ParDoPayload.class);
+    String mainInputId = null;
+    for (Map.Entry<String, String> input : ptransform.getInputsMap().entrySet()) {
```
Comment (kennknowles) [inline runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/ParDos.java]: `ptransform.getInputsMap().keys().removeAll(payload.getSideInputsMap().keys()`?

Comment (tgroh) [inline runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/ParDos.java]: Done.

File: runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/ParDos.java [hunk 2]
```diff
@@ -146,6 +148,20 @@ public static ParDoPayload toProto(ParDo.MultiOutput<?, ?> parDo, SdkComponents
     return doFnAndMainOutputTagFromProto(payload.getDoFn()).getMainOutputTag();
   }
 
+  public static RunnerApi.PCollection getMainInput(
```
Comment (kennknowles) [inline runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/ParDos.java]: This seems unused except in test. Have I missed something?

Should javadoc the requirement that it is a ParDo, or just decompose the input outside the method, and perhaps also make it private.

Comment (tgroh) [inline runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/ParDos.java]: This is going to be used within the Dataflow Worker

**Comment by kennknowles** [comment]
You know, I think I just reviewed one commit. Give me a second.

**Comment by kennknowles** [comment]
Yea the other commit is just fine. Great.

**Pull request closed without merging**